drivers/media/i2c/irs1125.c

```diff
@@ -921,6 +921,38 @@ static int irs1125_ctrls_init(struct irs1125 *sensor, struct device *dev)
 		goto error_ctrls;
 	}
 
+	for (i = 0; i < IRS1125_NUM_SEQ_ENTRIES; i++) {
+		char name[64];
+		struct v4l2_ctrl_config ctrl_cfg = {
+			.ops = &irs1125_ctrl_ops,
+			.id = IRS1125_CID_SAFE_RECONFIG_S0_EXPO + i * 2,
+			.type = V4L2_CTRL_TYPE_INTEGER,
+			.min = 0,
+			.max = U16_MAX,
+			.step = 1,
+			.def = 0x1000
+		};
+
+		snprintf(name, sizeof(name),
+			 "Safe reconfiguration of exposure of sequence %d", i);
+		ctrl_cfg.name = name;
+		ctrl = v4l2_ctrl_new_custom(hdl, &ctrl_cfg,
+					    NULL);
+		if (!ctrl)
+			dev_err(dev, "Failed to init custom control %s\n",
+				ctrl_cfg.name);
+
+		snprintf(name, sizeof(name),
```

Comment:
This has me worried.
```struct v4l2_ctrl``` only has a ```const char *name``` you're passing in a stack allocated pointer, and I don't believe anything in the framework is going to take a persistent copy of that. That means that ```name``` is pointer into random stuff, and is the same for all of your controls.

Does ```v4l2-ctl --list-ctrls``` actually list out the right thing?
I suspect not, and you really need two arrays of const char* names, with IRS1125_NUM_SEQ_ENTRIES entries in each. 

Doing that would also mean that it'd be neater to have two loops, one for adding the exposure ctrls, and one for the framerate ones. Initialise the ```struct v4l2_ctrl_config``` outside the loop, and then insert id and pointer to the name. i+=2 in the loop increment.

Reply:
You are right, the ctrl did not show up as expected. I changed this to use const char* arrays now.

I tested it again with `v4l2-ctl --lsit-ctrls` and it shows the expected results now.